### DISCUSSION TITLE
Is superadmin.p12 the same as tomcat.jks aka keystore.jks

### DISCUSSION THREAD

**Discussion by SolaLee** [Keyfactor/ejbca-ce#385] [General]
Dear Admin,

I have used EJBCA via Docker and enjoyed it, however for my Linux server I need to do the installation manually and NOT via Docker.

So begins my adventure of installing EJBCA into my Windows 11 before I proceed to the Linux server.

The many websites and this discussion group has helped me a lot, I would like to thank you.

Some technical information for someone reading this in the future.

I am running on Windows 11 Pro:
I am using :
- Java version is openjdk version "11.0.20" 2023-07-1
- Ant version is Apache Ant(TM) version 1.10.14 compiled on August 16 2023
- WildFly version is wildfly-preview-26.0.0.Final
- EJBCA version is from a EJBCA_8_0_20230531.tar.gz
- Database is psql (PostgreSQL) 12.9

Primary due to my non Ant background, a noob in WildFly and JBOSS and an idiot in PKI
I documented all my steps so when I had to redo the installation, I will remember what I did.

Finally I have managed to reached the end. I got 
- ant clean deloyear done (nervous waiting for the ejbca.ear to appear in the wildfly)
- ant runinstall (i was so happy after it passes the 2048 bit entry without error)
- finally I saw the tomcat.jks and truststore.jsk

Hmmm then i wonder, where is the superadmin.p12?

I continue running ant deploy-keystore, 
Ant copied the tomcat.jks from ejbca folder into my wildfly folder as keystore.jks 
Ant copied the truststore.jsk from ejbca folder into my wildfly folder as the same truststore.jsk

I opened up the keystore.jks and saw it has ManagementCA certificate + localhost certificate
I opened up the truststore.jks and saw it to contain only the ManagementCA certificate 

Can someone please correct but if I convert the keystore.jks into a keystore.p12, will that be the same as having the superadmin.p12?

My next journey is to get the https running so I can continue with accessing the fun feature of EJBCA.

**Comment by SolaLee** [comment]
I would like to answer my own question.
NO superadmin.p12 IS NOT THE SAME as keystore.jks

You are suppose to get tomcat.jks, truststore.jks and superadmin.p12 AFTER you run install.

If you dont please check your web.properties file the value superadmin.batch MUST BE true.
superadmin.batch=true

**Reply by gilles380** [reply]
ji @SolaLee 
If I may : 
I juste made native installation of ejbca/wildfly, deploy and run install ok
I have in my web.properties : 
Desired token type for the SSL server certificate:  P12, JKS. default value is JKS
httpsserver.tokentype=P12

now I tried importing my superadmin.p12 in firefox, it ask me to sign with pasword (defaut is ejbca) but then, nothing happens, the certificate is just NOT loaded but nor error...
If you have any clue ?

**Reply by SolaLee** [reply]
Dear Gilles380

Just to clarify on a few things

1. You have installed EJBCA which runs on Wildfly.
2. You have already installed the superadmin.p12 file in your other browser 
3. You are already able to view the EJBCA webpage with the other browser
Assuming you run on localhost at 8443,  you URL will be https://localhost:8443/ejbca/adminweb/

Now you are trying to do the same in firefox?
Am I correct?

**Reply by gilles380** [reply]
Hi thanks for replying. I finally managed to do the right thing, just an issue of linux file permission :)
Cheers

**Reply by SolaLee** [reply]
Fantastic... well done...  you checked the log and found the permission error.
